### ISSUE TITLE
Date Format Is Inconsistent? dd/mm/yyyy vs mm/dd/yyyy

### ISSUE THREAD

**Issue by Stickyhavr** [Stephen-Lon/Alfred-workflow-get-weekday-of-date#1]
Hello again,
I was testing this workflow out and it seems that the date format is off. The workflow asks for the dd/mm/yyyy format (international standard) for the input but it's actually taking the dd/mm/yyyy (US standard).  But only for day inputs less than 12.

In other words, if you put in 01/04/2023 it's going to tell you that January 4th 2023 is a Wednesday, not that April 1st is a Saturday!  However, if you ask it about 10/19/2023 it will complain about the range because it doesn't expect a number higher than 12 in the second set. So anyway, it seems like something is off for the input. 

As for the output, it would probably be more useful if it gave the date written out so that there's no possible confusion about whether it's talking about January 4th or April 1st... etc. So, for example: "January 4th 2023 is a Wednesday" or "4 January 2023 is a Wednesday" etc.

For bonus points: It might be nice if the workflow were configurable between mm/dd/yyyy & dd/mm/yyyy formats. I have no idea if that's easy to do or not though. Not a big deal either way as long as it's clear and works consistently.

Can you confirm this behavior on your end?  Thanks!

**Comment by Stephen-Lon** [comment]
I'm sorry for the inconvenience and have added a note to the Read Me. I strongly suspect it's because your system settings for the short form date do not match the dd/mm/yyyy format. If that's the case, even if you input a date in the correct format the workflow will not work because AppleScript will not recognise the input date format as being correct (i.e., as matching the short form date to which your system preferences are set).

I'm afraid there's currently no simple way (at least for me!) to output the date in the format you mention (i.e.,the longer form - because Alfred doesn't take two outputs from the AppleScript).

There may be a relatively simple hack to make the workflow work for you if your short system date format is mm/dd/yyyy. Try putting this in the Conditional Action as the regex string:

^(0[1-9]|1[0-2])\/(0[1-9]|[12][0-9]|3[01])\/\d{4}$

That will validate input in the mm/dd/yyyy format and your system settings should then simply recognise the short form date so that the AppleScript works accordingly. If you can confirm that works for you I'll update the workflow configuration with a note as to the alternative regex for those who use different short form date settings.

Stephen

**Comment by Stephen-Lon** [comment]
If you're able to confirm the work-around works I have an updated copy of the workflow to upload which you can configure, when importing it (or afterwards, using "Configure workflow"), to use the US date format simply by un-checking a check box in the configuration.

I can't test it completely because of my system settings.

Stephen

**Comment by Stickyhavr** [comment]
Oh, that makes perfect sense.
I can confirm that switching the RegEx in the conditional utility "Find weekday" does the trick. Thanks so much!

I tried to further troubleshoot and see if I could get it working with the original regex by changing my system short date format. But either that didn't work, or I didn't do it right, or maybe it require a reboot/logout to take effect.  What I did: went to Language & Region, then advanced, then the dates tab and dragged to switch the order of the mm and dd (on my system the example is June first, 2022 so I switched it so that it said 1 6 // 22.  But then the workflow still wasn't working using the dd/mm/yyyy format. 

Hold on, let me try again with a reboot.

**Comment by Stephen-Lon** [comment]
I'm just about to upload a completely new version which should do pretty well all you need (including the better output). I've not had the chance to test whether it works properly with U.S. date format so would appreciate your comments!

You can select the date format when configuring the workflow.

Stephen

**Comment by Stickyhavr** [comment]
Ok, even after a reboot it's not that. You have to actually change the region to make it work (and that takes effect right away, no need to reboot). Which makes total sense, I was just reluctant to change the region but it's an easy switch back.

So I can confirm that switching my region to the UK and the re-importing the workflow into Alfred (to reset it back to defaults) then does work as expected on my machine using the dd/mm/yyyy format (instead of mm/dd/yyyy).

I'll be happy to test the new version and report back.

Also, you might want to add a note in the workflow that if you are using a mm/dd region, you may want to change the text in two places:  the subtext of the keyword and the description of the input error. (Unless you've already thought of a better solution. I think it's reasonable to ask the user to make those two changes manually. Especially if you give them example text to copy and paste in the note).

**Comment by Stickyhavr** [comment]
Boom! I see you already fixed both of those text things by including both in the subtext and making the error text more general. And I love how the output now says the full date, or I guess the medium date. But you know what I mean. 

Happy to report workflow 2.0 beta is working as expected with US region (and checkbox unselected during import). :-)

**Comment by Stickyhavr** [comment]
> I'm sorry for the inconvenience

Not at all. Thank you so much for making all those improvements! I appreciate all of the workflows you've posted. I don't really have a need for "get characters" yet as that style of password verification is not yet common in the US (plus my password manager can display a character counter under the password for that purpose) but I downloaded it anyway just to have it handy in case I ever need it.

But relative dates, get weekday, and word count are all very useful to me. So thank you so much for making your workflows available and for responding to my original forum post, otherwise I might have never known these existed. I made a reminder in my to-do app to check your gitHub again in a few months just in case you end up posting any other great workflows. :-)

Much appreciated, thanks again!
